src/language/ast.ts

```diff
@@ -358,10 +375,37 @@ export interface FieldNode {
   readonly alias?: NameNode;
   readonly name: NameNode;
   readonly arguments?: ReadonlyArray<ArgumentNode>;
+  // Note: Client Controlled Nullability is experimental
+  // and may be changed or removed in the future.
+  readonly required?: ListNullabilityNode | NullabilityDesignatorNode;
+  readonly isInRequiredChain: Boolean;
```

Comment:
I think this will likely become a new required status instead since it feels like it's another case that isn't nullable or non nullable, but impacts null propagation rules all the same.